src/pip/_internal/cli/base_command.py

```diff
@@ -159,6 +160,14 @@ def main(self, args):
             logger.critical('ERROR: %s', exc)
             logger.debug('Exception information:', exc_info=True)
 
+            return ERROR
+        except BrokenStdoutLoggingError:
+            # Bypass our logger and write any remaining messages to stderr
+            # because stdout no longer works.
+            print('ERROR: Pipe to stdout was broken', file=sys.stderr)
```

Comment:
This entire block can probably be something along the lines of:

```py
logger.error("...", exc_info=self.verbosity > 0)
return ERROR
```

Reply:
@pradyunsg I had something like that earlier, but then I reconsidered because by this point in the except block, the logging framework has already thrown an exception. I didn't want to use a logging call again in case some handler is configured to use `stdout`. If it is, then that would trigger another exception while handling the original exception. So I thought it was better to bypass the logging module and print directly to `stderr` as a failsafe.